### PR TITLE
[dashboard] Implemented notification settings

### LINKED ISSUE #3398: [Dashboard] Notifications Page
### Problem to solve

Currently, notifications settings are mixed with other user settings like Environment Variables, API Tokens, etc.

This is part of https://github.com/gitpod-io/gitpod/issues/3300. 🍕 

### User Experience goal

The user should be able to a navigate and update their notifications settings while the system confirms their action by showing a toast notification.

### Proposal

In this first MVC (minimum viable change), we can do the following:

1. Introduce a dedicated notifications settings page.
2. Checking or unchecking the two existing options should trigger a toast notification at the bottom of the screen.

Components involved:
1. Checkbox
2. Toast

### Designs

| Notifications | Notifications (Spacing) |
|-|-|
| image | image |

In the designs above, there're are also hover states for the top tabs and the sidebar tabs.

See also [design specs](https://www.figma.com/file/tZXINzszBE9rzFvUVzTQa3/Notifications-3398).

### Documentation

No documentation changes are needed for this first iteration.

### PR DESCRIPTION
**Pull request by svenefftinge**
Fixes #3398

### CODE REVIEW

File: components/dashboard/src/settings/Notifications.tsx [hunk 1]
```diff
@@ -1,9 +1,62 @@
...
+            await getGitpodService().server.updateLoggedInUser({
+                allowsMarketingCommunication: user.allowsMarketingCommunication
+            });
+            ctx.setUser(user);
+        }
+    }
     return <div>
-        <SettingsPage title='Notifications' subtitle='Email notification preferences'>
-            <h3>Notifications</h3>
+        <SettingsPage title='Notifications' subtitle='Choose when to be notified'>
```
Comment (jankeromnes) [inline components/dashboard/src/settings/Notifications.tsx:38]:
```suggestion
        <SettingsPage title='Notifications' subtitle='Choose when to be notified.'>
```

**Review by gtsiolis — APPROVED** [review]
Smoooooth! 🍦 

Thanks @svenefftinge! The specs included a toast component for visual feedback on every change of the notifications settings, see https://github.com/gitpod-io/gitpod/issues/3398. Do we skip this for now?

File: components/dashboard/src/settings/Notifications.tsx [hunk 2]
```diff
@@ -1,9 +1,62 @@
...
+            <h3>Email Notification Preferences</h3>
```
Comment (jankeromnes) [inline components/dashboard/src/settings/Notifications.tsx:39]:
```suggestion
            <h3>Email notification preferences</h3>
```

File: components/dashboard/src/settings/Notifications.tsx [hunk 3]
```diff
@@ -1,9 +1,62 @@
...
+import { getGitpodService } from "../service/service";
+import { UserContext } from "../user-context";
 
 export default function Notifications() {
+    const ctx = useContext(UserContext);
+    const user = ctx.user;
+    const isTransactionalMail = !user?.additionalData?.emailNotificationSettings?.disallowTransactionalEmails;
+    const toggleTransactionalMail = async ()=>{
+        if (user) {
+            user.additionalData = {
```
Comment (jankeromnes) [inline components/dashboard/src/settings/Notifications.tsx:12]: Maybe get `user.additionalData` inside the handler?

Getting it outside of it means that you will overwrite any `additionalData` changes that happened between:
- This component was rendered
- A checkbox was ticked

Proof of concept: https://youtu.be/bYD8s3yivEw (my IDE change gets overwritten)

**Review by jankeromnes — APPROVED** [review]
Looks good to me!

Except the `user` should probably be fetched in the checkbox handler (otherwise you risk overwriting other `additionalData` changes, see below).

File: components/dashboard/src/settings/Notifications.tsx (hunk 1 above)
Comment (jankeromnes) [reply components/dashboard/src/settings/Notifications.tsx:38]: (Duplicate suggestion.)

File: components/dashboard/src/settings/Notifications.tsx (hunk 2 above)
Comment (gtsiolis) [reply components/dashboard/src/settings/Notifications.tsx:39]: I think we are moving forward with Title Case, right?

File: components/dashboard/src/settings/Notifications.tsx [hunk 4]
```diff
@@ -1,9 +1,62 @@
...
+                desc="Receive product marketing emails"
+                checked={isMarketingMail}
+                onChange={toggleMarketingMail}/>
         </SettingsPage>
     </div>;
 }
+
+function CheckBox(props: {title: string, desc: string, checked: boolean, onChange: () => void}) {
+    return <div className="flex mt-4">
+        <input className={"focus:ring-0 mt-1 rounded-sm cursor-pointer "+(props.checked?'bg-gray-800':'')} type="checkbox" checked={props.checked} onChange={props.onChange}/>
```
[the remark this answers is not in the document]
Comment (svenefftinge) [reply components/dashboard/src/settings/Notifications.tsx:56]: I tried it, but it somehow looks way too large. Maybe something else is off in relation.

File: components/dashboard/src/settings/Notifications.tsx (hunk 3 above)
Comment (svenefftinge) [reply components/dashboard/src/settings/Notifications.tsx:12]: The user is cached within the app. So in order to solve the (unlikely) case that someone changes settings in two browser tabs, we'd need to fetch the user from the server before altering it.

**Pull request merged**